pkg/test/resourcefixture/testdata/basic/sql/v1beta1/sqlinstance/sqlserverinstance/update.yaml

```diff
@@ -28,5 +28,6 @@ spec:
     sqlServerAuditConfig:
       bucketRef:
         name: storagebucket-${uniqueId}
+        kind: StorageBucket
```

Comment:
Yeah, but I think due to https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/1605/files, it's failing without kind specified. Seems like name + kind must be specified ..?

Reply:
I see where this comes from. I think we want to change the dependency from a generic ResourceRef to a more Kind specific format, and get rid of the `v1alpha1.ResourceRef`. So as in the future we can resolve and validate the actual reference to improve the dependency logics. 

You can follow this example to add new Kind specific reference under `./apis/refs`. (Feel free to do that in follow up PRs since this one is already pretty big).

Reply:
Ok, this may require an api change though..? Or, will existing resources continue to work if we change the type of the struct, as long as the field names stay the same?

Reply:
Hmm, I *think* it will be ok to switch to typed resource references, as long as no customer CRs previously specified the `kind` fields.